src/parseTools.js

```diff
@@ -436,18 +436,12 @@ function splitI64(value, floatConversion) {
   // For negatives, we need to ensure a -1 if the value is overall negative, even if not significant negative component
 
   const lowInput = legalizedI64s ? value : 'VALUE';
-  if (floatConversion) lowInput = asmFloatToInt(lowInput);
   const low = lowInput + '>>>0';
   const high = makeInlineCalculation(
-      asmCoercion('Math.abs(VALUE)', 'double') + ' >= ' + asmEnsureFloat('1', 'double') + ' ? ' +
-        '(VALUE > ' + asmEnsureFloat('0', 'double') + ' ? ' +
-        asmCoercion('Math.min(' + asmCoercion('Math.floor((VALUE)/' +
-        asmEnsureFloat(4294967296, 'double') + ')', 'double') + ', ' +
-        asmEnsureFloat(4294967295, 'double') + ')', 'i32') + '>>>0' +
+      'Math.abs(VALUE) >= 1 ? (VALUE > 0 ? ' +
+        'Math.min(Math.floor((VALUE)/4294967296), 4294967295)>>>0' +
```

Comment:
This is one place that I'm not sure I understand the implications of removing the coercions.

In this change I've removed all usage of `asmCoercion`.   I think all the other ones looks safe, but I don't grok this enough to know if it might have and effect.

Reply:
The `double` conversions (using `+(x)`) have no effect on doubles. But the `i32` one might be important. It might be easiest to print out the output of this to see exactly what the `|0` is on, for us to check.